book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,298 @@
+
+In Zebra the consensus related code lives in the `zebra-consensus` crate. The block subsidy checks are implemented in `zebra-consensus`, in the following module:
+
+- `zebra-consensus/src/block/subsidy/subsidy.rs` 
+
+Inside `zebra-consensus/src/block/subsidy/` the following submodules will be created:
+
+- `general.rs`: General block reward functions and utilities.
+- `founders_reward.rs`: Specific functions related to funders reward.
+- `funding_streams.rs`: Specific functions for funding streams.
+
+In addition to calculations the block subsidy requires constants defined in the protocol. The implementation will also create additional constants, all of them will live at:
+
+- `zebra-consensus/src/parameters/subsidy.rs`
+
+Checking functions for blocks are implemented at `zebra-consensus/src/block/check.rs` and they are called at `zebra-consensus/src/block.rs`. This follows the already existing structure for block validation in Zebra.
+
+It is important to note that all blocks before Sapling are verified by the CheckpointVerifier, so they are not considered in this design. The following table will show what periods are included and what is not in this proposal:
+
+| Height                                 | Miner    | Founder reward | Funding streams | Shielded Coinbase | Target Spacing |
+|----------------------------------------|----------|----------------|-----------------|-------------------|----------------|             
+| ~Genesis~                              | ~**0%**~ | ~**0%**~       | ~**0%**~        | ~**No**~          | ~**None**~     |
+| ~(Genesis + 1)..Sapling~               | ~80%~    | ~20%~          | ~0%~            | ~No~              | ~150 seconds~  |
+| Sapling..Blossom                       | 80%      | 20%            | 0%              | No                | 150 seconds    |
+| Blossom..Heartwood                     | 80%      | 20%            | 0%              | No                | 75 seconds     |
+| Heartwood..Canopy                      | 80%      | 20%            | 0%              | Yes               | 75 seconds     |
+| Canopy..Second Halving                 | 80%      | 0%             | 20%             | Yes               | 75 seconds     |
+| Second Halving..                       | 100%     | 0%             | 0%              | Yes               | 75 seconds     |
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Given the module structure proposed above, the block subsidy calculations from the protocol must be implemented. The final goal is to do semantic validation for each incoming block and make sure they pass the consensus rules.
+
+To do the calculations and checks the following constants, types and functions need to be introduced:
+
+## Constants
+
+### General
+
+- `SLOW_START_INTERVAL: Height`
+- `SLOW_START_SHIFT: Height`
+- `MAX_BLOCK_SUBSIDY: u64`
+- `PRE_BLOSSOM_POW_TARGET_SPACING: Duration`
+- `POST_BLOSSOM_POW_TARGET_SPACING: Duration`
```

Comment:
Let's turn these constants into a `block_target_spacing` method on `NetworkUpgrade` ?

We can make the `BLOSSOM_POW_TARGET_SPACING_RATIO` a fixed constant with the value `2`, because it isn't going to change.

I can make this change in PR #1256, because I need it there.

Reply:
Looks like a good idea however unsure if we should document such details here. Maybe we should add a sentence at the start of the Constants section like "Constants can be implemented as methods of types where appropriate"?

Reply:
So we don't need to specify these constants in this RFC.
```suggestion
```

Reply:
Because they are not used any more by the block subsidy checks - the block subsidy checks only use `BLOSSOM_POW_TARGET_SPACING_RATIO`.